server/controllers/vote.controller.js

```diff
@@ -77,37 +37,61 @@ function list(req, res, next) {
     .catch(e => next(e));
 }
 
-/**
- * Delete vote.
- * @returns {Vote}
- */
-function remove(req, res, next) {
-  const vote = req.vote;
-  vote.remove()
-    .then(deletedVote => res.json(deletedVote))
-    .catch(e => next(e));
+// To have the mobile clients start putting vote info insode of entityId
+// instead of postId. This will make vote.model more general.
+function movePostToEntity(req, res, next) {
+  if (req.post) {
+    req.entity = req.post;
+  }
+  next();
+}
+
+function findVote(req, res, next) {
+  // TODO: REMOVE once we migrate over to entityId only
+
+  const successCB = (vote) => {
+    if( vote) {
+      req.vote = vote;
+    }
+    next();
+  };
+
+  const errorCB = (error) => {
+    next(error);
+  };
+
+  // We need to account for the fact that posts could either be
+  // liked under entityId or postId
+  if (req.post) {
+    Vote.findOne({$or: [
+      {postId: req.post._id, userId: req.user._id},
```

Comment:
This can still be entity._id since req.post is being set to req.entity

Reply:
If you are checking for `req.post` you don't need this `$or`. You should abstract this to let the route load. So if you have a `/post/{id}/vote` route, the post controller should already load the correct entity.